### PR TITLE
Add SDAO - updated

### PR DESCRIPTION
**Pull request by RobertSDAO**

### CODE REVIEW

**Comment by RobertSDAO** [comment]
Updated the code

**Comment by llamatester** [comment]
Error while running adapter at projects/sdao/index.js: 
        
 
 
    
        Found export keys that were not part of specification: fetchTvl
    
        List of valid keys: 
    				
    				tvl
    				staking
    				methodology
    				pool2
    				misrepresentedTokens
    				fetch
    				timetravel
    				borrowed
    				start
    				masterchef
    				doublecounted
    				treasury
    				hallmarks
    				incentivized
    				offers
    				deadFrom

File: projects/sdao/index.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const fetchTvl = async function () {
+  const response = await fetch(
+    "https://api.singularitydao.ai/value/totalValueLocked"
```
Comment (g1nt0ki) [inline projects/sdao/index.js:3]: Can this be calculated using on-chain calls instead? We've got a guide on how to use our SDK here: https://docs.llama.fi/list-your-project/submit-a-project

**Comment by llamatester** [comment]
The adapter at projects/sdao/index.js exports TVL: 
        
 
 
    
    total                    40.93 M

**Comment by g1nt0ki** [comment]
HI, I am still seeing th tvl fetched from api instead of calculated from on chain data

**Comment by RobertSDAO** [comment]
> HI, I am still seeing th tvl fetched from api instead of calculated from on chain data

@g1nt0ki this is correct. Too many different moving pieces at this moment to calculate directly on-chain, including our dynasets (AI driven ETFs). 
Might update in near future. For now, TVL$, provided by our API is proficient as far as we are concerned.

**Comment by g1nt0ki** [comment]
> > HI, I am still seeing th tvl fetched from api instead of calculated from on chain data
> 
> @g1nt0ki this is correct. Too many different moving pieces at this moment to calculate directly on-chain, including our dynasets (AI driven ETFs). Might update in near future. For now, TVL$, provided by our API is proficient as far as we are concerned.

Hi @RobertSDAO, I looked into your dynasets, if you want, I can help you with creating tvl calculation for that, but when I add up the 3 dynasets, I only get less 3 million, the adapter returns 40M, how are you getting rest of the tvl?

**Comment by RobertSDAO** [comment]
> > > HI, I am still seeing th tvl fetched from api instead of calculated from on chain data
> > 
> > 
> > @g1nt0ki this is correct. Too many different moving pieces at this moment to calculate directly on-chain, including our dynasets (AI driven ETFs). Might update in near future. For now, TVL$, provided by our API is proficient as far as we are concerned.
> 
> Hi @RobertSDAO, I looked into your dynasets, if you want, I can help you with creating tvl calculation for that, but when I add up the 3 dynasets, I only get less 3 million, the adapter returns 40M, how are you getting rest of the tvl?

Hi @g1nt0ki . The rest comes from Liquidity pools, yield farms and various staking pools. 
Im the head of products and i told my sr developers i'll handle this (im a jr coder). 

I could try to use the sdk to get on chain data. Can i contact you on discord or anything? My handle there (im in the defillama discord channel is casartelli - robert)

**Comment by g1nt0ki** [comment]
I have trouble finding you, can you jump into defillama discord and write hi?

**Comment by g1nt0ki** [comment]
closing this in favor of https://github.com/DefiLlama/DefiLlama-Adapters/pull/2227

**Pull request closed without merging**